src/payment/strategies/quadpay/quadpay-payment-strategy.spec.ts

```diff
@@ -0,0 +1,262 @@
+import { getCheckout, getCheckoutState } from '../../../checkout/checkouts.mock';
+import { MissingDataError, RequestError } from '../../../common/error/errors';
+import { ContentType, INTERNAL_USE_ONLY } from '../../../common/http-request';
+import { getResponse } from '../../../common/http-request/responses.mock';
+import { getConfigState } from '../../../config/configs.mock';
+import { getCustomerState } from '../../../customer/customers.mock';
+import { OrderActionCreator, OrderActionType, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { getOrderRequestBody } from '../../../order/internal-orders.mock';
+import { PaymentMethodActionCreator, PaymentRequestSender } from '../../../payment';
+import { getPaymentMethodsState, getQuadpay } from '../../../payment/payment-methods.mock';
+import { RemoteCheckoutActionCreator, RemoteCheckoutRequestSender } from '../../../remote-checkout';
+import { createSpamProtection, PaymentHumanVerificationHandler } from '../../../spam-protection';
+import { StoreCreditActionCreator, StoreCreditActionType, StoreCreditRequestSender } from '../../../store-credit';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentActionType } from '../../payment-actions';
+import PaymentMethodRequestSender from '../../payment-method-request-sender';
+import { PaymentInitializeOptions } from '../../payment-request-options';
+import PaymentRequestTransformer from '../../payment-request-transformer';
+import { getErrorPaymentResponseBody } from '../../payments.mock';
+
+import QuadpayPaymentStrategy from './quadpay-payment-strategy';
+
+describe('QuadpayPaymentStrategy', () => {
+    let store: CheckoutStore;
+    let requestSender: RequestSender;
+    let orderActionCreator: OrderActionCreator;
+    let paymentActionCreator: PaymentActionCreator;
+    let paymentMethodActionCreator: PaymentMethodActionCreator;
+    let storeCreditActionCreator: StoreCreditActionCreator;
+    let remoteCheckoutActionCreator: RemoteCheckoutActionCreator;
+    let checkoutMock: Checkout;
+    let applyStoreCreditAction: Observable<Action>;
+    let submitOrderAction: Observable<Action>;
+    let submitPaymentAction: Observable<Action>;
+    let strategy: QuadpayPaymentStrategy;
+
+    beforeEach(() => {
+        const paymentClient = createPaymentClient(store);
+
+        store = createCheckoutStore({
+            checkout: getCheckoutState(),
+            customer: getCustomerState(),
+            config: getConfigState(),
+            cart: getCartState(),
+            paymentMethods: getPaymentMethodsState(),
+        });
+        requestSender = createRequestSender();
+
+        orderActionCreator = new OrderActionCreator(
+            paymentClient,
+            new CheckoutValidator(new CheckoutRequestSender(createRequestSender()))
```

Comment:
You can use variable declared on line #57

Reply:
good catch, thx!